frontend/pages/user/forgot-password.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FormEvent, useEffect, useState } from 'react';
+import { Montserrat } from 'next/font/google';
+
+import styles from '../../styles/User.module.css';
+import { doForgotPassword } from '../api/user';
+import LoadingOverlay from '../../components/LoadingOverlay';
+
+const montserrat = Montserrat({ subsets: ['latin'] });
+
+export default function Forgot_password() {
+
+  const [isLoading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [message, setMessage] = useState<string | null>(null);
+
+  const handleForgotPassword = (e: FormEvent<HTMLFormElement>) => {
+    setLoading(true);
+    e.preventDefault();
+
+    const reqData = {
+      email: e.currentTarget.userEmail.value,
+    } 
+
+    doForgotPassword(reqData, setMessage, setError).finally(() => setLoading(false));
+  };
+
+  const initForgotPassword = async () => {
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    setLoading(true);
+    initForgotPassword();
```

Comment:
This `useEffect` sets loading to true and to false immediately. Not sure why this was implemented.